ninja-core/src/main/java/ninja/utils/ImplFromPropertiesFactory.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ninja.utils;
+
+import com.google.inject.Injector;
+import org.slf4j.Logger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * An abstract provider that determines which implementation to load based on 
+ * the value of key in {@link NinjaProperties} (aka application.conf).
+ * 
+ * If this variable is set then the implementation class is loaded for subclasses
+ * to use in their own implementation of Provider.get().
+ */
+public class ImplFromPropertiesFactory<T> {
```

Comment:
Since it's used in 3 places that all have unit tests, I skipped including one.  I'll create a simple one today though just to be safe.